File: lab4/main.js
#!/usr/bin/env node
process.stdin.resume();
process.stdin.setEncoding('utf8');

var lingeringLine = "";

function my_printf(format_string,param){
	for(var i=0;i<format_string.length;i++){
		if((format_string.charAt(i) == '#') && (format_string.charAt(i+1) == 'k')){
			process.stdout.write(param);
			i++;
		}else{
			process.stdout.write(format_string.charAt(i));
		}
	}
	console.log("");
}

process.stdin.on('data', function(chunk) {
	lines = chunk.split("\n");

	lines[0] = lingeringLine + lines[0];
	lingeringLine = lines.pop();
	for(var i=0;i<lines.length;i++){
		my_printf(lines[i],lines[i+1])
		i++;
	}

});

File: lab4/main.py
#!/usr/bin/env python3

import sys
def print_digits(format_string,index):
    buffer=""
    for i in range (index,len(format_string)):
        if format_string[i].isdigit():
            buffer += format_string[i]
        elif format_string[i]!=" ":
            return ""
        else:
            return buffer[::-1]   
    return ""


def my_printf(format_string,param):
    #print(format_string)
    shouldDo=True
    buffer=""
    for idx in range(0,len(format_string)):
        if shouldDo:
            if format_string[idx] == '#' and format_string[idx+1]=='g':
                print(param,end="")
                buffer=print_digits(format_string,idx+3)
                if(buffer!=""):
                    print(" "+buffer,end="")
                shouldDo=False     
            else:
                if (buffer==""):
                    print(format_string[idx],end="")
        else:
            buffer==""
            shouldDo=True
    print("")




data=sys.stdin.readlines()

for i in range(0,len(data),2):
    my_printf(data[i].rstrip(),data[i+1].rstrip())
